vitamins/scs_bearing_block.scad

```diff
@@ -0,0 +1,147 @@
+            polygon(points = [ [0,0], [base, 0], [top, height], [0, height] ]);
+    }
+
+    boltHoleRadius = screw_clearance_radius(S1);
+    footHeight = min(0.75, (G - bearing_dia(scs_bearing(type)) - 1.5) / 2); // estimate, not specified on drawings
+
+    color(sks_bearing_block_color) {
+        linear_extrude(L, center = true) {
+            bearingRadius = bearing_dia(scs_bearing(type)) / 2;
+            // center section with bearing hole
+            difference() {
+                union() {
+                    translate([-(B - 2 * boltHoleRadius) / 2, -h + footHeight])
+                        square([B - 2 * boltHoleRadius, G - footHeight]);
+                    for(m = [0, 1])
+                        mirror([m, 0, 0])
+                            translate([0 , G - h])
+                                right_trapezoid(bearingRadius, bearingRadius -F + G, F - G);
+                }
+                circle(r = bearingRadius);
+            }
+            // add the sides
+            for(m = [0, 1])
+                mirror([m, 0, 0]) {
+                    trapezoidX = boltHoleRadius - 0.5; // estimate, not specified on drawings
+                    sideX = 2 * (K - boltHoleRadius - trapezoidX);
+                    chamfer = 0.5;
+                    assert(sideX > chamfer, "trapezoidX too large in scs_bearing_block");
+                    translate([B / 2 + boltHoleRadius, -h]) {
+                        square([sideX - chamfer, G]);
+                        translate([sideX, 0])
+                            rotate(90)
+                                right_trapezoid(G - chamfer, G, chamfer);
+                        translate([sideX, 0]) {
+                            right_trapezoid(trapezoidX - footHeight, trapezoidX, footHeight);
+                            translate([trapezoidX, footHeight])
+                                rotate(90)
+                                    right_trapezoid(T - footHeight, L1 - footHeight, trapezoidX);
+                        }
+                    }
+                    translate([B / 2 - boltHoleRadius, -h])
+                        mirror([1, 0, 0])
+                            right_trapezoid(boltHoleRadius,  boltHoleRadius + footHeight, footHeight);
+                }
+        }
+        // side blocks with bolt holes
+        for(x = [-B / 2, B / 2])
+            translate([x, G / 2 - h, 0])
+                rotate([90, 0, 0])
+                    linear_extrude(G, center = true)
+                        difference() {
+                            square([boltHoleRadius * 2, L], center = true);
+                            for (y = [-C / 2, C / 2])
+                                translate([0, y])
+                                    circle(r = boltHoleRadius);
+                        }
+    }
+    not_on_bom() no_explode()
+        linear_bearing(scs_bearing(type));
+}
+
+
+module scs_bearing_block_assembly(type, screw_length, screw_type) { //! Assembly with screws and nuts in place
```

Comment:
I don't understand why the screw length is specified or why the nuts are placed a washer's thickness from the end of the screw. All other assemblies like this are passed a panel thickness. The screw length is then calculated and the nut and washer would be offset downwards by the panel thickness.

Reply:
Screw length is specified so that the bearing block can be attached to items of varying thickness. I hadn't thought of passing the panel thickness instead - I'll do that.

Reply:
A simple example here: https://github.com/nophead/NopSCADlib/blob/master/printed/door_hinge.scad#L159